### DISCUSSION TITLE
Welcome to dietaryindex Discussions!

### DISCUSSION THREAD

**Discussion by jamesjiadazhan** [jamesjiadazhan/dietaryindex#3] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Bianlab2023** [comment]
First of all, thank you for your contribution. I am also calculating the relevant index. when i did function AHEI_NHANES_FPED，i found the score of alcohol=0 was calculated as AHEI_ALCOHOL = 0. in fact, it was 2.5 according to the reference.Obviously, you made a mistake in using the case_when function. At the same time, in my opinion, calculating the index involving deciles or medians without taking into account complex sampling weights may lead to bias.

**Reply by jamesjiadazhan** [reply]
Thank you for your feedback. Apparently, the AHEI function in NHANES has the issue you refer to, which is not found in other AHEI functions after my investigation. I found the error is due to the inappropriate order of the conditions in the case_when function. Now the bug is fixed.

Regarding complex sampling weights, you don't need to consider when calculating the dietary index. You can always consider sampling weights later as shown below:
```
#set up survey design 
design <- svydesign(id = ~SDMVPSU, 
                    strata = ~SDMVSTRA, 
                    weight = ~WTDRD1, 
                    data = merged_nhanes, #set up survey design on the full dataset #can restrict at time of analysis 
                    nest = TRUE) 


#(1) this model tests if HEI score per survey cycle (using categorical SDDSRVYR) is statistically different from 1999-2000 (reference group) 
model_1_a <- svyglm(HEI2015_ALL.x ~ SDDSRVYR + age + sex + race + education, 
                            subset(design, recallstatus=='1')) #apply survey design and subset to just high quality dietary results
```

**Reply by Bianlab2023** [reply]
Thanks for reply. Sorry for confusion about the complex sampling weights issues, like sodium in AHEI ,you used the decile directly without considering complex sampling weights. Also,  can you give the reference about SSB_FNDDS_1718 data file?

**Reply by jamesjiadazhan** [reply]
I don't have a reference for the SSB file, as I manually selected all possible beverages, including soda, coffee, tea, with any added sugar, summarized them in the individual food file (FPED_IND file), and then converted them to tsp. I selected all beverages with any added sugar because it is objective. Usually, people don't explain this process, and they selected some food codes for representing SSB, while certain food codes with less added sugar (e.g. milk tea or Frappuccino) may be ignored.

**Reply by jamesjiadazhan** [reply]
Theoretically, you should repeat this process for other years since food codes change every year.
